design-documents/graph-ql/coverage/catalog/configurable-options-selection.graphqls

```diff
@@ -1,21 +1,29 @@
 type ConfigurableProduct {
-    configurable_options_selection_metadata(selectedConfigurableOptionValues: [ID!]): ConfigurableOptionsSelectionMetadata @doc(description: "Metadata for the specified configurable options selection")
+    configurable_product_options_selection(configurableOptionValues: [ID!]): ConfigurableProductOptionsSelection @doc(description: "Specified configurable product options selection")
+    variants: [ConfigurableVariant] @deprecated(reason: "Use configurable_options_selection_metadata instead.") @doc(description: "An array of variants of products") @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\ConfigurableVariant")
+    configurable_options: [ConfigurableProductOptions] @deprecated(reason: "Use configurable_options_selection_metadata instead.") @doc(description: "An array of linked simple product items") @resolver(class: "Magento\\ConfigurableProductGraphQl\\Model\\Resolver\\Options")
 }
 
-type ConfigurableOptionsSelectionMetadata @doc(description: "Metadata corresponding to the configurable options selection.")
+type ConfigurableProductOptionsSelection @doc(description: "Metadata corresponding to the configurable options selection.")
 {
-    options_available_for_selection: [ConfigurableOptionAvailableForSelection!] @doc(description: "Configurable options available for further selection based on current selection.")
+    configurable_options: [ConfigurableProductOption!] @doc(description: "Configurable options available for further selection based on current selection.")
     media_gallery: [MediaGalleryInterface!] @doc(description: "Product images and videos corresponding to the specified configurable options selection.")
     variant: SimpleProduct @doc(description: "Variant represented by the specified configurable options selection. It is expected to be null, until selections are made for each configurable option.")
 }
 
-type ConfigurableOptionAvailableForSelection @doc(description: "Configurable option available for further selection based on current selection.") {
-    option_value_uids: [ID!]! @doc(description: "Configurable option values available for further selection.")
-    attribute_code: String! @doc(description: "Attribute code that uniquely identifies configurable option.")
+type ConfigurableProductOption {
+    uid: ID!
+    attribute_code: String!
+    attribute_uid: ID!
+    label: String!
+    sort_order: Int!
+    values: [ConfigurableProductOptionValue!]
 }
 
-# Configurable option value type has to be extended to include ID which can be used to uniquely identify the option value across the system. This is consistent with proposal of single mutation for add-to-cart
-type ConfigurableProductOptionsValues {
+type ConfigurableProductOptionValue {
     uid: ID!
-    is_available_for_selection: Boolean!
+    is_available: Boolean! # when display OOS is enabled, if the variant selection is OOS, this will return false.
```

Comment:
what is it 'OOS'? can you provide full name please?

Reply:
I was referring to Out of stock. 
<img width="935" alt="Screen Shot 2020-10-22 at 1 49 18 PM" src="https://user-images.githubusercontent.com/43612864/96916684-7f9e3e80-146d-11eb-848d-8bf7e2841812.png">

Will update the doc.